### ISSUE TITLE
`geohex_grid` aggregation results missing `doc_count_error_upper_bound` and `sum_other_doc_count` fields

### ISSUE THREAD

**Issue by philvarner** [opensearch-project/geospatial#304]
**What is the bug?**

`geohex_grid` aggregation results missing `doc_count_error_upper_bound` and `sum_other_doc_count` fields.

**How can one reproduce the bug?**

1. Index data with a geo_point field (named `centroid` in the example below).
2. Run geohex_grid aggregation like:
```
POST /my_index/_search
{
  "aggs": {
    "grid_geohex_frequency": {
      "geohex_grid": {
        "field": "centroid",
        "precision": 3
      }
    }
  },
  "size": 0
}
```

The response is like:

```
"aggregations": {
    "grid_geohex_frequency": {
      "buckets": [ ... 10000 buckets ] 
  }
}
```

**What is the expected behavior?**

I would expect the response to also have `doc_count_error_upper_bound` and `sum_other_doc_count` fields when there are documents which are not covered by by the buckets returned, as is done with term aggregation.

**What is your host/environment?**

AWS OpenSearch Service 2.5

**Do you have any screenshots?**

n/a

**Do you have any additional context?**

no

**Comment by VijayanB** [comment]
I believe those parameters are part of term aggregation. Geo hex aggregation will consider all documents unlike terms where you can use size parameter to list only top buckets.

**Comment by philvarner** [comment]
geohex_grid also has a size parameter [per the documentation](https://opensearch.org/docs/2.4/opensearch/geohexgrid-agg/), which defaults to 10000, so this is exactly like terms.

The geoaggregations should be consistent with terms, since they're just another type of bucket value. Not having these fields, especially sum_other_doc_count, limits the usefulness of geoaggregations, since you have to use a heuristic (bucket count is exactly 10000) to determine if there were documents that are not represented in the bucket values and you'd have to run separate count of determine how many documents were not represented.

**Comment by VijayanB** [comment]
@philvarner You are absolutely right. Thanks for detailed explanation and reporting this bug, we will look into this issue. However, feel free to contribute if you know how to fix this issue.

**Comment by VijayanB** [comment]
@nknize This is not available with other geo aggregations like geo_hash and geo_tile. I believe this can be added to BaseGeoGrid, so that all geo aggregations can inherit this value in response. Do you see an issue in enabling this feature for geo aggregations conceptually? Please share your feedback. Thanks
